docs/source/llms/openai/notebooks/openai-quickstart.ipynb

```diff
@@ -0,0 +1,505 @@
+    "3. The OpenAI SDK. It's [available on PyPI](https://pypi.org/project/openai/) here. For this tutorial, we're going to be using version 0.28.1 (the last release prior to the 1.0 release).\n",
+    "\n",
+    "To install the `openai` SDK library that is compatible with this notebook to try this out yourself, as well as the additional `tiktoken` dependency that is required for the MLflow integration with `openai`, simply run:\n",
+    "\n",
+    "```bash\n",
+    "pip install 'openai<1' tiktoken\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### A Note on Key Management Best Practices\n",
+    "\n",
+    "API keys serve as critical access points to services and, as such, need to be handled with utmost care and security. They are akin to digital keys that unlock access to valuable data, services, and, importantly, financial resources. In the context of Software-as-a-Service (SaaS) Large Language Models (LLMs), API keys are directly linked to billing and charge based on the volume of requests made. Hence, these keys are as sensitive as your credit card information.\n",
+    "\n",
+    "#### Best Practices for Handling API Keys:\n",
+    "\n",
+    "- **Keep Them Secret**: Treat API keys like your most sensitive passwords. Never expose them in public repositories, forums, or websites.\n",
+    "- **Limit Exposure**: Avoid hardcoding them in your code. Instead, use environment variables or secure key management services.\n",
+    "- **Rotate Them**: Periodically change your API keys to mitigate risks associated with key leakage.\n",
+    "\n",
+    "### Setting Your API Key Safely\n",
+    "\n",
+    "The safest way to use your API key is by setting it as an environment variable. This method prevents accidental exposure and makes the key readily available to your applications.\n",
+    "\n",
+    "**For macOS and Linux**:\n",
+    "\n",
+    "1. Open your terminal.\n",
+    "2. Edit your shell’s profile script (~/.bashrc for Bash or ~/.zshrc for Zsh).\n",
+    "3. Add the following line:\n",
+    "\n",
+    "```bash\n",
+    "export OPENAI_API_KEY='your_openai_api_key_here'\n",
+    "```\n",
+    "4. Save the file and run `source ~/.bashrc` (or `source ~/.zshrc`) to reload the profile script.\n",
+    "\n",
+    "This will set the key for all future terminal sessions.\n",
+    "\n",
+    "**For Windows**:\n",
+    "\n",
+    "1. **Search** for \"Environment Variables\" in your Start menu and select \"Edit the system environment variables.\"\n",
+    "2. In the System Properties dialog, click the \"Environment Variables...\" button.\n",
+    "3. Under \"User variables,\" click \"New...\" to create a new environment variable.\n",
+    "4. Set \"Variable name\" as `OPENAI_API_KEY` and \"Variable value\" as your actual API key.\n",
+    "5. Click OK and restart any open command lines or applications that need to use this variable."
+   ]
```

Comment:
This section sounds like we need to update `~/.bashrc` or `~/.zshrc`.


<img width="1006" alt="Screenshot 2023-12-12 at 19 55 37" src="https://github.com/mlflow/mlflow/assets/17039389/45266ac9-d157-4d7e-97b8-1a5ea62bf285">

Reply:
Reduced the content here by a lot and just added links to MSFT and Apple docs pages for setting ENV VARs to keep this section as light as possible.